server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/SharingParentMailboxWithoutChildren.feature

```diff
@@ -0,0 +1,75 @@
+#***************************************************************
+# Licensed to the Apache Software Foundation (ASF) under one   *
+# or more contributor license agreements.  See the NOTICE file *
+# distributed with this work for additional information        *
+# regarding copyright ownership.  The ASF licenses this file   *
+# to you under the Apache License, Version 2.0 (the            *
+# "License"); you may not use this file except in compliance   *
+# with the License.  You may obtain a copy of the License at   *
+#                                                              *
+#   http://www.apache.org/licenses/LICENSE-2.0                 *
+#                                                              *
+# Unless required by applicable law or agreed to in writing,   *
+# software distributed under the License is distributed on an  *
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+# KIND, either express or implied.  See the License for the    *
+# specific language governing permissions and limitations      *
+# under the License.                                           *
+# **************************************************************/
+Feature: Share parent mailbox without sharing submailbox
+  As a James user
+  I want to be able to retrieve my mailboxes
+
+  Background:
+    Given a domain named "domain.tld"
+    And a user "alice@domain.tld"
+    And a user "bob@domain.tld"
+    And "alice@domain.tld" has a mailbox "shared"
+    And "alice@domain.tld" has a mailbox "caca"
+    And "alice@domain.tld" has a mailbox "shared.secret"
+    And "alice@domain.tld" shares her mailbox "shared" with "bob@domain.tld" with "aeilrwt" rights
+    And "alice@domain.tld" has a message "m1" in "shared" mailbox
+    And "alice@domain.tld" has a message "m2" in "shared.secret" mailbox
+    And "alice@domain.tld" has a message "m3" in "caca" mailbox
+
+  Scenario: Bob can only see shared mailbox but not shared/secret mailbox
+    When "bob@domain.tld" lists mailboxes
+    Then the mailboxes should contain "shared" in "Delegated" namespace
+    And the mailboxes should not contain "shared.secret" in "Delegated" namespace
+    And the mailboxes should not contain "secret" in "Delegated" namespace, with parent mailbox "shared" of user "alice@domain.tld"
+    And the mailboxes should not contain "secret" in "Personal" namespace
+    And the mailbox "shared" has 1 messages
+
+  Scenario: Alice can see shared mailbox and shared/secret mailbox
+    When "alice@domain.tld" lists mailboxes
+    And the mailboxes should contain "shared" in "Personal" namespace
+    And the mailboxes should contain "secret" in "Personal" namespace, with parent mailbox "shared" of user "alice@domain.tld"
+    And the mailbox "shared" has 1 messages
+    And the mailbox "secret" has 1 messages
+
+  Scenario: Alice can get message from shared and shared/secret mailbox
```

Comment:
I don't think it's a good pattern to have multiple When/Then in a single scenario.

Reply:
Not here anymore anyway